cg/cli/analysis/balsamic.py

```diff
@@ -34,36 +41,79 @@ def balsamic(context, case_id, priority, email, target_bed):
     context.obj['analysis_api'] = AnalysisAPI
     context.obj['fastq_handler'] = BalsamicFastqHandler
     context.obj['gzipper'] = gzip
+    context.obj['tb'] = tb.TrailblazerAPI(context.obj)
+
+    hk_api = hk.HousekeeperAPI(context.obj)
+    scout_api = scoutapi.ScoutAPI(context.obj)
+    lims_api = lims.LimsAPI(context.obj)
+    context.obj['api'] = AnalysisAPI(db=context.obj['db'],
+                                     hk_api=hk_api,
+                                     tb_api=context.obj['tb'],
+                                     scout_api=scout_api,
+                                     lims_api=lims_api,
+                                     deliver_api=None)
     if context.invoked_subcommand is None:
         if case_id is None:
-            LOGGER.error('provide a case')
+            LOG.error('provide a case')
             context.abort()
 
         # execute the analysis!
-        context.invoke(link, family_id=case_id)
+        context.invoke(link, case_id=case_id)
         context.invoke(config, case_id=case_id, target_bed=target_bed)
-        context.invoke(run, run_analysis=True, case_id=case_id, priority=priority, email=email)
+        context.invoke(run,
+                       run_analysis=True,
+                       case_id=case_id,
+                       priority=priority,
+                       email=email)
 
 
 @balsamic.command()
-@click.option('-d', '--dry-run', 'dry', is_flag=True, help='print config to console')
+@click.option('-c', '--case', 'case_id', help='link all samples for a case')
+@click.argument('sample_id', required=False)
+@click.pass_context
+def link(context, case_id, sample_id):
+    """Link FASTQ files for a SAMPLE_ID."""
+
+    link_objs = get_links(context, case_id, sample_id)
+
+    for link_obj in link_objs:
+        LOG.info("%s: %s link FASTQ files", link_obj.sample.internal_id,
+                 link_obj.sample.data_analysis)
+        if link_obj.sample.data_analysis and 'balsamic' in link_obj.sample.data_analysis.lower(
+        ):
+            LOG.info('%s has blasamic as data analysis, linking.',
+                     link_obj.sample.internal_id)
+            context.obj['api'].link_sample(BalsamicFastqHandler(context.obj),
+                                           case=link_obj.family.internal_id,
+                                           sample=link_obj.sample.internal_id)
+        else:
+            LOG.warning(
+                '%s does not have blasamic as data analysis, skipping.',
+                link_obj.sample.internal_id)
+
+
```

Comment:
In this part, I removed `click.abort()` since for a particular sample: `setkoi` due to a RNA sample it failed to link Normal sample. 

The check to mixed samples should be handled before reaching this stage, and order form/portal/code should take care of it.

Reply:
"The check to mixed samples should be handled before reaching this stage, and order form/portal/code should take care of it." What's your idea of how that should be taken care of?

Reply:
Through the order portal/form maybe? I think there are too many logical statements to keep track of them.

If a sample is RNA and BALSAMIC, it should be linked anyway. A mixed case with a sample only with MIP in its analysis tag shouldn't end up in Balsamic analysis anyway. If it is tagged as such, it should have been picked up during previous steps and addressed there.

Reply:
[There is no "Orderportal"](https://www.youtube.com%2Fwatch%3Fv%3DeBQXwJHM6lI&usg=AOvVaw2y17_UOwR36s3syzkFhuzh). Cg *IS* the Orderportal.